### PR TITLE
Update to dunfell

### PR DESCRIPTION
**Pull request by macpijan**
This addresses the point one from: https://github.com/brocaar/chirpstack-gateway-os/pull/54#issuecomment-633061008 

Currently, build for `rasspberrypi3` is running on my build machine.

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -27,4 +27,4 @@
 	branch = zeus
 [submodule "layers/meta-rust-bin"]
 	path = layers/meta-rust-bin
-	url = https://github.com/rust-embedded/meta-rust-bin.git
+	url = https://github.com/3mdeb/meta-rust-bin.git
```
Comment (macpijan) [inline .gitmodules:30]: This can be changed back to rust-embedded as soon as this gets merged: https://github.com/rust-embedded/meta-rust-bin/pull/71

**Comment by macpijan** [comment]
Build for raspberrypi3 passed. I have no way to test it on hardware currently.

**Comment by brocaar** [comment]
Thanks @macpijan, this is a great contribution :+1: Especially since now we can build the Go binaries from source, so the hack to let the recipe download the binary is not needed anymore.

Lets wait until your PR has been merged into the meta-rust-bin repository. It should be straightforward to review and merge. I have commented on your PR in the hope this will be picked up soon :-)

**Comment by toepi** [comment]
should this pr not split? dunfell and build from source?

**Comment by macpijan** [comment]
Could be. I can do that if needed.

File: .gitmodules (hunk 1 above)
Comment (brocaar) [reply .gitmodules:30]: It looks like the other submodules are still at the zeus branch. Is that correct?

**Comment by brocaar** [comment]
@toepi @macpijan I don't mind combining these. The main reason to upgrade the Yocto version is to facilitate building the Go binaries from source.

Without the latter, the first would not be needed but without the first, the latter would not be possible.

File: meta/conf/layer.conf [hunk 2]
```diff
@@ -8,7 +8,4 @@ BBFILES += "${LAYERDIR}/recipes-*/*/*.bb \
 BBFILE_COLLECTIONS += "chirpstack-gateway-os"
 
 BBFILE_PATTERN_chirpstack-gateway-os = "^${LAYERDIR}/"
-LAYERSERIES_COMPAT_chirpstack-gateway-os = "zeus"
-
-# Required by swupdate
-PREFERRED_PROVIDER_u-boot-fw-utils = "libubootenv"
```
Comment (brocaar) [inline meta/conf/layer.conf:14]: Why is this removed + the `u-boot-fw-utils_%.bbappend` ?

Comment (brocaar) [reply meta/conf/layer.conf:14]: Nevermind.

File: .gitmodules (hunk 1 above)
Comment (macpijan) [reply .gitmodules:30]: @brocaar Hmm, I guess I'm not sure what is the impact of this field there. When I clone the repo and update the modules (`make submodules`) they all do have correct revisions (from dunfell, not from zeus).

Comment (macpijan) [reply .gitmodules:30]: I can update the branches in the `.gitmodules` file as well.

**Comment by brocaar** [comment]
I'm currently testing this.

**Comment by brocaar** [comment]
Thanks @macpijan for your work. I have merged this pull-request with some minor modifications:

* I updated all the meta-layers to their latest (dunfell) versions
* I added a fix to compile Wireguard
